### PR TITLE
create `randomvariable` module

### LINKED ISSUE #392: `randomvariable` module
I think there are enough concrete but generic `RandomVariable`s that they should have their own module rather than living in `metaclass.py`, especially since they aren't actually abstract

Candidates for this include `DeterministicVariable`, `DistributionalRV`, `TransformedRandomVariable`, etc.

^ This list also makes clear that we should also standardize their suffixes (versus having all of `Variable`, `RandomVariable`, and `RV`.

### PR DESCRIPTION
**Pull request by sbidari**
- [x] moves `DistributionalRV` and `TransformedRandomVariable` from `metaclass` to `randomvariable` module,
- [x] renames
  -  `DistributionalRV` -> `DistributionalVariable` 
  -  `StaticDistributionalRV` -> `StaticDistributionalVariable` 
  -  `DynamicDistributionalRV` -> `DynamicDistributionalVariable` 
  - `TransformedRandomVariable` -> `TransformedVariable`

closes #392

### CODE REVIEW

**Comment by damonbayer** [comment]
@dylanhmorris @sbidari Do we need to make any changes about what is in the `deterministic` module?

**Comment by sbidari** [comment]
> @dylanhmorris @sbidari Do we need to make any changes about what is in the `deterministic` module?

Potentially move `DeterministicVariable` to the `randomvariable` module? @dylanhmorris mentioned it in the issue comment.
But since `DeterministicVariables` are not really random variables, it might make more sense to leave them in the deterministic module?

**Review by damonbayer — APPROVED** [review]

**Comment by damonbayer** [comment]
We agreed not to move the `DeterministicVariable` class at this time.

**Comment by dylanhmorris** [comment]
@sbidari is this ready for review?

File: pyrenew/latent/hospitaladmissions.py
Comment (damonbayer) [inline pyrenew/latent/hospitaladmissions.py:217]: Looks like this has unapproved/unmerged changes from https://github.com/CDCgov/PyRenew/pull/356. These should be removed or we should wait until https://github.com/CDCgov/PyRenew/pull/356 is merged.

Comment (sbidari) [reply pyrenew/latent/hospitaladmissions.py:217]: #356 is ready for review, let's wait until it's merged

**Comment by sbidari** [comment]
> @sbidari is this ready for review?

Waiting for #356, I unintentionally created this branch with changes from #356

**Comment by damonbayer** [comment]
@sbidari https://github.com/CDCgov/PyRenew/pull/356 is merged. Please mark this as "ready for review" when  it's ready for review.

File: pyrenew/metaclass.py
Comment (damonbayer) [inline pyrenew/metaclass.py:125]: This appears to be out of date.

**Review by damonbayer — CHANGES_REQUESTED** [review]
Looks like the merge undid some changes from https://github.com/CDCgov/PyRenew/pull/356.

File: pyrenew/metaclass.py
Comment (sbidari) [reply pyrenew/metaclass.py]: removed in https://github.com/CDCgov/PyRenew/pull/412/commits/e1b888af1989c3d605a79507522ecaa583640958

**Review by damonbayer — APPROVED** [review]
Thanks @sbidari!

**Review by damonbayer — CHANGES_REQUESTED** [review]
My approval was premature. The `randomvariable` module documentation needs to be added to the website.

**Review by damonbayer — APPROVED** [review]

**Pull request merged**
